js-weather/assets/javascript/weather_fetcher.js

```diff
@@ -0,0 +1,53 @@
+export class Wether {
+  constructor(city) {
```

Comment:
cityの引数は、コンストラクタでなく`fetchByXMLHttpRequest`や`fetchByAsyncAwait`の引数として渡すのが良いと思います。

そうすれば、Weatherクラスの初期化（newの呼び出し）を1回で済ませられます。

```suggestion
  constructor() {
```